src/command.rs

```diff
@@ -77,255 +77,268 @@ pub async fn init(kind: PackageType, name: Option<PackageName>) -> miette::Resul
-    let manifest = Manifest::read().await?;
-    let package = PackageStore::current()?.release(manifest).await?;
+        let mut manifest = Manifest::read().await?;
 
-    let path = directory.as_ref().join(format!(
-        "{}-{}.tgz",
-        package.manifest.package.name, package.manifest.package.version
-    ));
+        manifest
+            .dependencies
+            .push(Dependency::new(registry, repository, package, version));
 
-    if !dry_run {
-        fs::write(path, package.tgz)
+        manifest
+            .write()
             .await
-            .into_diagnostic()
-            .wrap_err(miette!(
-                "failed to write package release to the current directory"
-            ))?;
+            .wrap_err(miette!("failed to write `{MANIFEST_FILE}`"))
     }
 
-    Ok(())
-}
+    /// Removes a dependency from this project
+    pub async fn remove(self: Arc<Self>, package: PackageName) -> miette::Result<()> {
+        let mut manifest = Manifest::read().await?;
 
-/// Publishes the api package to the registry
-pub async fn publish(
-    credentials: Credentials,
-    registry: RegistryUri,
-    repository: String,
-    #[cfg(feature = "git")] allow_dirty: bool,
-    dry_run: bool,
-) -> miette::Result<()> {
-    #[cfg(feature = "git")]
-    if let Ok(repository) = git2::Repository::discover(Path::new(".")) {
-        let statuses = repository
-            .statuses(None)
-            .into_diagnostic()
-            .wrap_err(miette!("failed to determine repository status"))?;
+        let match_idx = manifest
+            .dependencies
+            .iter()
+            .position(|d| d.package == package)
+            .ok_or_else(|| miette!("package {package} not in manifest"))?;
 
-        if !allow_dirty && !statuses.is_empty() {
-            tracing::error!("{} files in the working directory contain changes that were not yet committed into git:\n", statuses.len());
+        let dependency = manifest.dependencies.remove(match_idx);
 
-            statuses
-                .iter()
-                .for_each(|s| tracing::error!("{}", s.path().unwrap_or_default()));
+        // if PackageStore::uninstall(&dependency.package).await.is_err() {
+        //     tracing::warn!("failed to uninstall package {}", dependency.package);
+        // }
 
-            tracing::error!("\nTo proceed with publishing despite the uncommitted changes, pass the `--allow-dirty` flag\n");
+        PackageStore::current()?
+            .uninstall(&dependency.package)
+            .await
+            .ok(); // temporary due to broken test
 
-            bail!("attempted to publish a dirty repository");
-        }
+        manifest.write().await
     }
 
-    let artifactory = Artifactory::new(registry, &credentials)?;
-
-    let manifest = Manifest::read().await?;
-    let package = PackageStore::current()?.release(manifest).await?;
+    /// Packages the api and writes it to the filesystem
+    pub async fn package(
```

Comment:
I structutally dislike the change regarding to the commands.rs, it creates a weird asymmetry between different commands

Reply:
I'm not sure I understand what you mean, can you elaborate?

Reply:
See my other comment :) The init command is now in a weird place and there are more potential cases for this

Reply:
I've addressed it in the other comment.